actions/rules/action_state_to_metastate.go

```diff
@@ -0,0 +1,467 @@
+			if columnRuleKey == ActionKeyStateToMetastate {
+				// unmarshall the configuration.
+				config := map[string]string{}
+				err := json.Unmarshal([]byte(columnRuleConfig), &config)
+				if err != nil {
+					return nil, nil, err
+				}
+				if metaState, ok := config[ActionKeyStateToMetastateConfigMetastate]; ok {
+					if metaState == wi.Fields[workitem.SystemMetaState] && !alreadyPlacedInColumn {
+						// the column config matches the *new* metastate, so the WI needs to
+						// appear in this column.
+						var currentSystemBoardColumn []interface{}
+						currentSystemBoardColumn, ok = wi.Fields[workitem.SystemBoardcolumns].([]interface{})
+						// flip alreadyPlacedInColumn, so this will be the new column. Other
+						// matching columns are ignored on this board.
+						alreadyPlacedInColumn = true
+						if !ok && wi.Fields[workitem.SystemBoardcolumns] != nil {
+							// again, wi.Fields[workitem.SystemBoardcolumns] may be empty, so we
+							// only fail here when the conversion fails AND the slice is non nil.
+							return nil, nil, errs.New("error converting SystemBoardcolumns set")
+						}
+						if !act.contains(currentSystemBoardColumn, column.ID.String()) {
+							wi.Fields[workitem.SystemBoardcolumns] = append(currentSystemBoardColumn, column.ID.String())
+						}
+						columnsChanged = true
+					} else {
+						// the column *does not* match the *new* metastate, so the column has to
+						// be removed from the WIs columns. Note that we can't just remove all
+						// entries in wi.Fields[workitem.SystemBoardcolumns] as there may be
+						// other columns from non-relevant boards in it that need to be left
+						// untouched.
+						var currentSystemBoardColumn []interface{}
+						currentSystemBoardColumn, ok = wi.Fields[workitem.SystemBoardcolumns].([]interface{})
+						if !ok && wi.Fields[workitem.SystemBoardcolumns] != nil {
+							// again, wi.Fields[workitem.SystemBoardcolumns] may be empty, so we
+							// only fail here when the conversion fails AND the slice is non nil.
+							return nil, nil, errs.New("error converting SystemBoardcolumns set")
+						}
+						wi.Fields[workitem.SystemBoardcolumns] = act.removeElement(currentSystemBoardColumn, column.ID.String())
+						columnsChanged = true
+					}
+				} else {
+					return nil, nil, errs.New("invalid configuration for transRuleKey '" + ActionKeyStateToMetastate + "': " + columnRuleConfig)
+				}
+			}
+		}
+	}
+	// if the column set has changed, create an entry for the change set.
+	if columnsChanged {
+		changes = act.addOrUpdateChange(changes, workitem.SystemBoardcolumns, oldColumnsConfig, wi.Fields[workitem.SystemBoardcolumns])
+	}
+	// finally, store the new work item state if something changed.
+	var resultingWorkItem workitem.WorkItem
+	if wiDirty {
+		result, err := act.storeWorkItem(&wi)
+		resultingWorkItem = *result
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+	// and return to sender.
+	return resultingWorkItem, changes, nil
```

Comment:
We would be returning an empty workitem in `resultingWorkitem` if there were no changes. Is this expected?

Reply:
Good find. Actually no, because the issue is somewhere else, but connected to this. In the case of the state change, `isDirty` is always true at l465. So this works, but for the wrong reasons. I'll update the method.

Reply:
973fe2d. Checked again, indeed the `isDirty` was redundant. I think this happened because I copied and refactored the `onColumnChange()` method there. Again, good find.